src/lib/vasi-sync/tests/atomic-tls-map-tests.rs

```diff
@@ -0,0 +1,213 @@
+//! This file contains tests intended to be run using [loom]. See the
+//! [crate-level documentation](crate) for details about running these under
+//! loom.
+//!
+//! [loom]: <https://docs.rs/loom/latest/loom/>
+
+mod sync;
+
+mod atomic_tls_map_tests {
+    use std::num::NonZeroUsize;
+
+    use super::sync;
+
+    use vasi_sync::atomic_tls_map::AtomicTlsMap;
+
+    #[test]
+    fn test_empty() {
+        sync::model(|| {
+            let _table = AtomicTlsMap::<10, u32>::new();
+        })
+    }
+
+    #[test]
+    fn test_store_and_load() {
+        sync::model(|| {
+            let table = AtomicTlsMap::<10, u32>::new();
+            unsafe {
+                assert_eq!(
+                    *table.get_or_insert_with(NonZeroUsize::try_from(1).unwrap(), || 11),
+                    11
+                );
+                assert_eq!(
+                    *table.get_or_insert_with(NonZeroUsize::try_from(2).unwrap(), || 12),
+                    12
+                );
+
+                // `get_or_insert_with` again with the same keys should return the previous
+                // values.
+                assert_eq!(
+                    *table.get_or_insert_with(NonZeroUsize::try_from(1).unwrap(), || 1000),
+                    11
+                );
+                assert_eq!(
+                    *table.get_or_insert_with(NonZeroUsize::try_from(2).unwrap(), || 1000),
+                    12
+                );
+
+                // Likewise `get` should return the inserted values.
+                assert_eq!(
+                    table
+                        .get(NonZeroUsize::try_from(NonZeroUsize::try_from(1).unwrap()).unwrap())
```

Comment:
Is the extra `try_from` needed here? It seems to be converting a `NonZeroUsize` to a `NonZeroUsize`?

Reply:
Oops, done